sdk/identity/azure-identity/src/client_credential_core.cpp

```diff
@@ -6,24 +6,29 @@
 #include "private/tenant_id_resolver.hpp"
 #include "private/token_credential_impl.hpp"
 
-#include <utility>
-
 using Azure::Identity::_detail::ClientCredentialCore;
 
 using Azure::Core::Url;
 using Azure::Core::Credentials::TokenRequestContext;
 using Azure::Identity::_detail::TenantIdResolver;
 using Azure::Identity::_detail::TokenCredentialImpl;
 
-decltype(ClientCredentialCore::AadGlobalAuthority) ClientCredentialCore::AadGlobalAuthority
-    = "https://login.microsoftonline.com/";
+namespace {
+const std::string AadGlobalAuthority = "https://login.microsoftonline.com/";
+}
 
+// The authority host used be the credentials is in the following order of precedence:
+// 1. AuthorityHost option set/overriden by the user.
+// 2. The value of AZURE_AUTHORITY_HOST environment variable, which is the default value of the
+// option.
+// 3. If the option is empty, use Azure Public Cloud.
 ClientCredentialCore::ClientCredentialCore(
     std::string tenantId,
     std::string const& authorityHost,
     std::vector<std::string> additionallyAllowedTenants)
     : m_additionallyAllowedTenants(std::move(additionallyAllowedTenants)),
-      m_authorityHost(Url(authorityHost)), m_tenantId(std::move(tenantId))
+      m_authorityHost(Url(authorityHost.empty() ? AadGlobalAuthority : authorityHost)),
```

Comment:
If you GA like this, it will be technically a breaking change to address #5002 after GA.

With this being released, you could do:
```
Options options;
options.AuthorityHost = "";
Credential(options); // would use public cloud
```

Reply:
Agreed. We are going to reject empty authority host, as an error.